File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/TaskColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TaskRow } from "./TaskRow";
import { TaskStatus } from "./TaskStatus";

export interface TaskColumns {
    TaskId: Column<TaskRow>;
    Title: Column<TaskRow>;
    AssignedToFullName: Column<TaskRow>;
    AssignedByUsername: Column<TaskRow>;
    DueDate: Column<TaskRow>;
    Status: Column<TaskRow>;
}

export class TaskColumns extends ColumnsBase<TaskRow> {
    static readonly columnsKey = 'Operations.Task';
    static readonly Fields = fieldsProxy<TaskColumns>();
}

[TaskStatus]; // referenced types

File: HRMS/HRMS.Web/Modules/Operations/Task/TaskColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using HRMS.Operations;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.Task")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int TaskId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string AssignedToFullName { get; set; }
    public string AssignedByUsername { get; set; }
    public DateTime DueDate { get; set; }
    [Width(100), QuickFilter, AlignCenter]
    public TaskStatus Status { get; set; }
}
